src/main/java/pl/javastart/couponcalc/PriceCalculator.java

```diff
@@ -1,13 +1,63 @@
 package pl.javastart.couponcalc;
 
+import java.math.BigDecimal;
+import java.math.RoundingMode;
 import java.util.List;
 
 public class PriceCalculator {
 
     public double calculatePrice(List<Product> products, List<Coupon> coupons) {
         // TODO
+        if (products == null || products.isEmpty()) {
+            return 0;
+        }
 
-        return 0;
+        double priceTotalSum;
+
+        if (coupons == null || coupons.isEmpty()) {
+            priceTotalSum = calculateTotalPriceIfNoCoupons(products);
+        } else if (coupons.size() == 1) {
+            Coupon coupon = coupons.get(0);
+            priceTotalSum = calculateTotalPriceForOneCoupon(products, coupon);
```

Comment:
Ten przypadek jest niepotrzebny. To jest to samo co niżej (>1 kupon), po prostu pętla wykona się raz.